src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1125,80 +1125,86 @@ public void downloadMailArchiveZipFile(Environment env, HttpServletRequest reque
-
-					log.info("begin: start downloading mail archive zip file for site " + site_id);
-
-					downloadStatus = getMailArchiveZipContent(env, site_id, downloadStatus,
-							sessionId, httpContext, out);
-
-					out.flush();
-					out.close();
-					log.info("Finished mail archive zip download for site " + site_id);
-
-
-				} catch (RestClientException e) {
-					String errorMessage = Utils.STATUS_FAILURE + " Cannot find site by siteId: " + site_id
-							+ " " + e.getMessage();
-					Response.status(Response.Status.NOT_FOUND).entity(errorMessage)
-					.type(MediaType.TEXT_PLAIN).build();
-					log.error(errorMessage);
-					downloadStatus.put("status", errorMessage);
-				}
-				catch (IOException e) {
-					String errorMessage = Utils.STATUS_FAILURE + " Problem getting mail archive zip file for "
-							+ site_id + " " + e.getMessage();
-					Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-					.entity(errorMessage).type(MediaType.TEXT_PLAIN)
-					.build();
-					log.error(errorMessage);
-					downloadStatus.put("status", errorMessage);
-
-				} catch (Exception e) {
-					String errorMessage = Utils.STATUS_FAILURE + " Migration status for " + site_id + " "
-							+ e.getClass().getName();
-					Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-					.entity(errorMessage).type(MediaType.TEXT_PLAIN)
-					.build();
-					log.error("downloadMailArchiveZipFile ", e);
-					downloadStatus.put("status", errorMessage);
-				}
-			} else {
-				String userError = "Cannot become user " + userId;
-				log.error(userError);
-				downloadStatus.put("status", userError);
-			}
-
-			// update the status and end_time of migration record
-			setMigrationEndTimeAndStatus(migrationId, repository, downloadStatus);
-
-			return;
-
+            // hold download status
+
+            JSONObject downloadStatus = new JSONObject();
+
+            // login to CTools and get sessionId
+            if (sessionAttributes.containsKey(Utils.SESSION_ID)) {
+                String sessionId = (String) sessionAttributes.get(Utils.SESSION_ID);
+                HttpContext httpContext = (HttpContext) sessionAttributes
+                        .get("httpContext");
+                Map<String, String[]> parameterMap = request.getParameterMap();
+                String destination_type = parameterMap.get("destination_type")[0];
+                try {
+                    //
+                    // Sends the response back to the user / browser. The
+                    // content for zip file type is "application/zip". We
+                    // also set the content disposition as attachment for
+                    // the browser to show a dialog that will let user
+                    // choose what action will he do to the sent content.
+                    //
+                    response.setContentType(Utils.MIME_TYPE_ZIP);
+                    String zipFileName=null;
+                    if(destination_type.equals("mailarchivezip")){
```

Comment:
Extract method to reduce duplication.  Even really short methods are worthwhile since they mean less room for oversight when making updates and, more important, less code that someone reading the code has to understand.